.github/workflows/deploy.yml

```
@@ -0,0 +1,119 @@
+name: Build and deploy Docker and Helm
+on:
+  push:
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+' # semver stable
+      - 'v[0-9]+.[0-9]+.[0-9]+-*' # semver with prerelease suffix
+jobs:
+  metadata:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+    outputs:
+      version: ${{ steps.metadata.outputs.version }}
+      release-exist: ${{ steps.metadata.outputs.release_exist }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: Metadata
+        id: metadata
+        env:
+          tag: ${{ github.ref_name }}
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          set -Eeuo pipefail
+
+          version=${tag#v}
+          release_exist="true"
+
+          if ! gh release view v${version} 1> /dev/null ; then
+            release_exist="false"
+          fi
+
+          echo "version="$version"" >>"$GITHUB_OUTPUT"
+          echo "release_exist="$release_exist"" >>"$GITHUB_OUTPUT"
+  deploy-docker:
```

nitpick, newlines ftw
```suggestion
          echo "release_exist="$release_exist"" >>"$GITHUB_OUTPUT"
          
  deploy-docker:
```